### PR TITLE
Matched FileWillBeUploaded Signature to Defined in Interface

### LINKED ISSUE #125: FileWillBeUploaded does not implement the server hook
The `FileWillBeUploaded` hook in [file_hooks.go](https://github.com/mattermost/mattermost-plugin-demo/blob/master/server/file_hooks.go) is never called by the server, as it does not implement the function defined by the interface.

The plugin's function signature is:
```go
FileWillBeUploaded(c *plugin.Context, fileInfo *model.FileInfo, reader bytes.Reader, buf *bytes.Buffer)
```

but the one defined in [the interface](https://github.com/mattermost/mattermost-server/blob/master/plugin/hooks.go#L193) is:

```go
FileWillBeUploaded(c *Context, info *model.FileInfo, file io.Reader, output io.Writer) (*model.FileInfo, string)
```

We need to change the signature to match the one defined by the interface and check that it actually gets called when a file is uploaded.

### PR DESCRIPTION
**Pull request by adithyaakrishna**
Signed-off-by: Adithya Krishna <aadithya794@gmail.com>

#### Summary
Updates FileWillBeUploaded function signature to be similar to what's defined in the interface

#### Ticket Link
This PR Fixes #125

### CODE REVIEW

File: server/file_hooks.go [hunk 1]
```diff
@@ -46,4 +46,5 @@ func (p *Plugin) FileWillBeUploaded(c *plugin.Context, fileInfo *model.FileInfo,
 			)
 		}
 	}
+	return nil, "" // <--- This is the new return line
```
Comment (agarciamontoro) [inline server/file_hooks.go:49]: Can we remove the comment in this line? I wrote it in my example simply as a guide.

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Thanks, @adithyaakrishna! Just one more thing I commented below and we're good to go :rocket:

**Comment by agarciamontoro** [comment]
Hi, @adithyaakrishna! Is there anything we can do to help you with this?

**Comment by adithyaakrishna** [comment]
@agarciamontoro Oops! I had missed that one, I am working on it rn and will send a commit regarding this soon :)

File: server/file_hooks.go (hunk 1 above)
Comment (adithyaakrishna) [reply server/file_hooks.go]: Thanks for pointing that out :)

**Review by hanzei — APPROVED** [review]
Nice :+1:

**Comment by adithyaakrishna** [comment]
Yayyy 🎉

**Pull request merged**